datafusion/common/src/scalar.rs

```diff
@@ -726,6 +726,24 @@ impl ScalarValue {
         }
     }
 
+    /// whether this value is true or not.
+    pub fn is_true(&self) -> bool {
+        match self {
+            ScalarValue::Boolean(v) => v.eq(&Some(true)),
```

Comment:
We can make more use of pattern matching here:

```suggestion
            ScalarValue::Boolean(&Some(true)) => true,
```